### ISSUE TITLE
"envDirName" does not work ("dotenvDirname" does)

### ISSUE THREAD

**Issue by jg-cantaa** [AnWeber/vscode-httpyac#93]
The envDirName Setting does not work, although it is the option documented to work everywhere.
VSCode doesn't even recognize the option as valid.
![Screenshot_2021-12-08_14-24-52](https://user-images.githubusercontent.com/95756912/145216013-0c9ad24c-c57a-4cae-be23-319d3eb398b4.png)
![Screenshot_2021-12-08_14-24-26](https://user-images.githubusercontent.com/95756912/145216020-4c6f34ae-ce67-4b93-baa7-878056ac8dcf.png)

**Comment by AnWeber** [comment]
I think the cause is the typo in the path of `httpyac.envDirName`. I search for [folder `env` by default](https://github.com/AnWeber/httpyac/blob/main/src/store/httpFileStore.ts#L131). The search starts from folder of the http file and searches for [the following files/ folder](https://github.com/AnWeber/httpyac/blob/main/src/store/httpFileStore.ts#L126-L132) upwards (package.json, .httpyac.js, httpyac.config.json, env). However, since `envDirName` is now overwritten with `.spec/env`, it cannot find root and read the files. Commenting out `envDirName` fixes the error and it will find the files again.

**Comment by jg-cantaa** [comment]
So, after doing some experimenting I am still quite confused. Although the issue title does not apply, I am still having difficulties getting this to work. So I will try to explain how my current understanding from experimenting is:

Assuming I have the folder structure
```
.vscode/
    settings.json
specs/
    api/
        request.http
    env/
        .env.test

```
- It searches for the envDirName Folder first in the following places and order:
  - `specs/api/<envDirName>`
  - `specs/<envDirName>`
  - `<envDirName>`
- If it finds a matching folder, It does not try to continue searching for other matching folders.
- specifying `envDirName` with a absolute path will work 
- if the path is not absolute, it seems to have trouble resolving paths with more than one segment (e.g. Specifying `envDirName` to be `test/env` only finds environments in the first `test` folder it finds.) 

In all honesty, this seems overly complicated to me, although I might not have the full picture and there might be valid reasons for this. The Documentation in the readme states "relative or absolute path to folder with dotenv files", which from my experimentation does not hold.

Could this be adjusted to reflect the description given in the readme?

**Comment by AnWeber** [comment]
The issue for the root folder determination is https://github.com/AnWeber/vscode-httpyac/issues/50. Use case is that different environment variables can be set for different requests in the same project. Many Tools like [tsc](https://github.com/microsoft/TypeScript/blob/325435891accdf3235d5fa0fc02c4af7a4d6da68/src/server/editorServices.ts#L1896), [eslint](https://eslint.org/docs/user-guide/configuring/configuration-files#using-configuration-files) or [Prettier](https://prettier.io/docs/en/configuration.html) also support such a config search. 
And if you use it in the standard way, your example would just work without changing `envDirName`.

> if the path is not absolute, it seems to have trouble resolving paths with more than one segment (e.g. Specifying envDirName to be test/env only finds environments in the first test folder it finds.)

Nested `envDirName` I have not considered correctly, and overlooked during implementation But I would recommend to just use the default one. I left the setting mainly for absolute file paths to allow cross-project settings or as hidden folder support in Linux.

**Comment by AnWeber** [comment]
fix is released with v4.8.2. Thanks:-)
